Feat/folder creation modal layout

alexanderleegs opened the pull request:
This PR introduces a folder creation modal, to resolve https://github.com/isomerpages/isomercms-frontend/issues/343. Saving functionality will be introduced in a separate PR. In addition, this PR also swaps the existing redirection to use a custom hook, `useRedirectHook`.

The folder/subfolder creation modal consists of two portions. Initially, the user is asked to select a folder name:
<img width="897" alt="Screenshot 2021-02-26 at 11 16 37 AM" src="https://user-images.githubusercontent.com/22111124/109250147-1caa2300-7824-11eb-83b4-f7051e3891fb.png">

Then the user chooses pages from the existing unlinked pages/pages in the folder to add to the new folder/subfolder:
<img width="1231" alt="Screenshot 2021-02-26 at 11 17 24 AM" src="https://user-images.githubusercontent.com/22111124/109250205-377c9780-7824-11eb-8efd-5a098a821759.png">

The pages can be sorted as well. Currently, only sorting by name and order in the collection.yml file are supported.

gweiying commented on src/components/FolderCreationModal.jsx:
```diff
@@ -0,0 +1,237 @@
+    }
+  }
+
+  return (
```
Discussed offline with Alex, some changes we could consider:

- Splitting up this FolderCreationModal into two files so that its more readable, 1) screen modal for folder name, which can be reused for editing folder name in future 2) screen for selecting files
- Letting redirection be handled in Folders.jsx rather than in FolderCreationModal.jsx, since both Folder creation and Page creation will require redirection.

kwajiehao commented on src/components/FolderCreationModal.jsx:
```diff
@@ -0,0 +1,237 @@
+  const sortOptions = [
+    ... parentFolder 
+      ? [{
+        value: 'collection',
+        label: 'Collection',
```
Since we've agreed to move away from using the term `Collection`, can we use another term for the default order?

alexanderleegs replied:
Good point, fixed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/f1a2924858d087d7fbdc0468fbf3b9ed30541bf6

kwajiehao reviewed (changes requested):
Spotted several UI bugs:

1. There shouldn't be text next to the button
<img width="1370" alt="Screenshot 2021-03-02 at 3 01 21 PM" src="https://user-images.githubusercontent.com/31984694/109612315-826f1580-7b6a-11eb-955e-1b478004821c.png">

2. Text color on hover should be white
<img width="466" alt="Screenshot 2021-03-02 at 3 01 26 PM" src="https://user-images.githubusercontent.com/31984694/109612386-9dda2080-7b6a-11eb-8dbc-261a92918b91.png">

3. Unable to see all pages in the page-selection modal (ura-lkywcp-4.0 repo)
<img width="1515" alt="Screenshot 2021-03-02 at 3 03 24 PM" src="https://user-images.githubusercontent.com/31984694/109612439-af232d00-7b6a-11eb-8aff-f2c8ec953702.png">

4. Selection button should be next to the `Sort by` text. 
<img width="387" alt="Screenshot 2021-03-02 at 3 07 44 PM" src="https://user-images.githubusercontent.com/31984694/109612509-cb26ce80-7b6a-11eb-81d9-fb01373b8b3c.png">

Re: the appearance, we could theoretically overwrite the CSS of the `react-select` library but we can leave that for later.

alexanderleegs commented:
> 1. There shouldn't be text next to the button
> 
> <img alt="Screenshot 2021-03-02 at 3 01 21 PM" width="1370" src="https://user-images.githubusercontent.com/31984694/109612315-826f1580-7b6a-11eb-955e-1b478004821c.png">

Fixed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/da701b8ebea756093b5eb0b60d5318d59f3bfabc

> 1. Text color on hover should be white
> 
> <img alt="Screenshot 2021-03-02 at 3 01 26 PM" width="466" src="https://user-images.githubusercontent.com/31984694/109612386-9dda2080-7b6a-11eb-8dbc-261a92918b91.png">

Fixed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/c0807cec6c88c36ff21b293cac5b75aedddb4b6d and https://github.com/isomerpages/isomercms-frontend/pull/359/commits/18207eef6131ad7c763d5d6545d68f31fdfbf02a

> 1. Unable to see all pages in the page-selection modal (ura-lkywcp-4.0 repo)
> 
> <img alt="Screenshot 2021-03-02 at 3 03 24 PM" width="1515" src="https://user-images.githubusercontent.com/31984694/109612439-af232d00-7b6a-11eb-8aff-f2c8ec953702.png">

Fixed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/595a9c1b0d51d649ea041fbf1e8196798da11bcf

> 1. Selection button should be next to the `Sort by` text.
> 
> <img alt="Screenshot 2021-03-02 at 3 07 44 PM" width="387" src="https://user-images.githubusercontent.com/31984694/109612509-cb26ce80-7b6a-11eb-81d9-fb01373b8b3c.png">
> 
> Re: the appearance, we could theoretically overwrite the CSS of the `react-select` library but we can leave that for later.

Resolved in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/6b802bd6a84724c80097a7606681b10f816e617f

> Splitting up this FolderCreationModal into two files so that its more readable: 1) screen modal for folder name (can be reused for editing folder name in future) 2) screen for selecting files

Changed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/51259826a3aa9a3b3a23bc571f375b81b221ef3b

> Letting redirection be handled in Folders.jsx rather than in FolderCreationModal.jsx, since both Folder creation and Page creation will require redirection.

Changed in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/ecea4090a29041cc2c3f497d9eb552f547c9c70a

kwajiehao commented on src/hooks/useRedirectHook.jsx:
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect, useState } from "react"
+import { Redirect, useHistory } from "react-router-dom"
```
unused imports `React` and `Redirect`

alexanderleegs replied:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/b910e4654fdd2a7fc17860197412921b364d5be3

kwajiehao commented on src/layouts/Workspace.jsx:
```diff
@@ -20,6 +19,7 @@ import { prettifyPageFileName } from '../utils';
 
 // Import hooks
 import useSiteColorsHook from '../hooks/useSiteColorsHook';
+import useRedirectHook from '../hooks/useRedirectHook';
```
unused import

alexanderleegs replied:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/b910e4654fdd2a7fc17860197412921b364d5be3

kwajiehao commented on src/hooks/useRedirectHook.jsx:
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect, useState } from "react"
+import { Redirect, useHistory } from "react-router-dom"
```
unused imports `React` and `Redirect`

alexanderleegs replied:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/b910e4654fdd2a7fc17860197412921b364d5be3

kwajiehao commented on src/components/FolderCreationModal.jsx:
```diff
@@ -26,11 +27,10 @@ const FolderCreationModal = ({
 }) => {
   // Errors
   const [errors, setErrors] = useState('')
+  const { setShouldRedirect, setRedirectUrl } = useRedirectHook()
```
unused `useRedirectHook` - is this here to anticipate future use?

alexanderleegs replied:
Yeah, will add in when save functionality is added

kwajiehao commented on src/components/FolderCreationModal.jsx:
```diff
@@ -26,11 +27,10 @@ const FolderCreationModal = ({
   pagesData,
   siteName,
   setIsFolderCreationActive,
-  setRedirectToNewPage,
```
update proptypes for this component to remove `setRedirectToNewPage`, `setNewPageUrl`

alexanderleegs replied:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/359/commits/522b998ef4cf177516b136488f58a9abf882989c

kwajiehao reviewed (changes requested):
Left comments on unused imports

kwajiehao reviewed (approved):
lgtm
